src/map/skill.c

```diff
@@ -13910,28 +13910,51 @@ static int skill_unit_onplace_timer(struct skill_unit *src, struct block_list *b
 		case UNT_ZENKAI_LAND:
 		case UNT_ZENKAI_FIRE:
 		case UNT_ZENKAI_WIND:
-			if( battle->check_target(&src->bl,bl,BCT_ENEMY) > 0 ){
-				switch( sg->unit_id ){
-					case UNT_ZENKAI_WATER:
+			if (battle->check_target(&src->bl, bl, BCT_ENEMY) > 0) {
+				switch (sg->unit_id) {
+				case UNT_ZENKAI_WATER:
+					switch (rnd()%3) {
+					case 0:
 						sc_start(ss, bl, SC_COLD, sg->val1*5, sg->skill_lv, skill->get_time2(sg->skill_id, sg->skill_lv));
+						break;
+					case 1:
 						sc_start(ss, bl, SC_FREEZE, sg->val1*5, sg->skill_lv, skill->get_time2(sg->skill_id, sg->skill_lv));
+						break;
+					case 2:
 						sc_start(ss, bl, SC_FROSTMISTY, sg->val1*5, sg->skill_lv, skill->get_time2(sg->skill_id, sg->skill_lv));
 						break;
-					case UNT_ZENKAI_LAND:
+					}
+					break;
+				case UNT_ZENKAI_LAND:
+					switch (rnd()%2) {
```

Comment:
```suggestion
					switch (rnd() % 2) {
```

https://github.com/HerculesWS/Hercules/wiki/Coding-Style#31--spaces

> Use one space around (on each side of) most binary and ternary operators,
> such as any of these:
> 
> ```
> 	=  +  -  <  >  *  /  %  |  &  ^  <=  >=  ==  !=  ?  :
> ```